### PR TITLE
feat: Adapt OZ V5

### PR DESCRIPTION
**Pull request by huyhuynh3103**
### Checklist
- [x] I have clearly commented on all the main functions following the [NatSpec Format](https://docs.soliditylang.org/en/v0.8.0/natspec-format.html)
- [x] The box that allows repo maintainers to update this PR is checked
- [ ] I tested locally to make sure this feature/fix works

### CODE REVIEW

File: src/ERC721Common.sol [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
   { }
 
   /// @inheritdoc IERC721State
-  function stateOf(uint256 _tokenId) external view virtual override returns (bytes memory) {
-    require(_exists(_tokenId), "ERC721Common: query for non-existent token");
-    return abi.encodePacked(ownerOf(_tokenId), nonces[_tokenId], _tokenId);
+  function stateOf(
+    uint256 tokenId
+  ) external view virtual override returns (bytes memory) {
+    _requireOwned(tokenId);
```
Comment (TuDo1403) [inline src/ERC721Common.sol:19]: I far as I remember, this do return owner if not throw error

File: src/ERC721Common.sol [hunk 2]
```diff
@@ -66,17 +75,14 @@ contract ERC721Common is ERC721Nonce, ERC721PresetMinterPauserAutoIdCustomized,
...
-    returns (uint256[] memory _tokenIds)
-  {
+  function bulkMint(
+    address[] calldata _recipients
+  ) external virtual onlyRole(MINTER_ROLE) returns (uint256[] memory tokenIds) {
     require(_recipients.length > 0, "ERC721Common: invalid array lengths");
-    _tokenIds = new uint256[](_recipients.length);
+    tokenIds = new uint256[](_recipients.length);
 
     for (uint256 _i = 0; _i < _recipients.length; _i++) {
```
Comment (TuDo1403) [inline src/ERC721Common.sol:84]:
```suggestion
    for (uint256 i; i < _recipients.length; i++) {
```

Comment (TuDo1403) [inline src/ERC721Common.sol:84]:
```suggestion
    for (uint256 i = 0; i < recipients.length; i++) {
```

File: src/ERC1155Common.sol
Comment (TuDo1403) [inline src/ERC1155Common.sol:1]: where is pause function 🔫

Comment (huyhuynh3103) [reply src/ERC1155Common.sol:1]: It does have pause functions, just don't show up in the git diff because there are no changes on this

**Review by TuDo1403 — APPROVED** [review]

**Pull request merged**